src/constraints/StringConstraints.ts

```diff
@@ -44,3 +58,78 @@ export function stringLengthNe(length: number): IConstraint<string> {
 	const expected = `expected.length !== ${length}`;
 	return stringLengthComparator(ne, 's.string.lengthNe', expected, length);
 }
+
+export function stringEmail(): IConstraint<string> {
+	return {
+		run(input: string) {
+			return validateEmail(input)
+				? Result.ok(input)
+				: Result.err(new ConstraintError('s.string.email', 'Invalid email address', input, 'expected to be an email address'));
+		}
+	};
+}
+
+export function stringRegex(regex: RegExp, type: 'uuid' | 'regex', expected?: string): IConstraint<string> {
+	expected ??= `expected to match ${type === 'regex' ? regex.source : `a ${type}`}`;
```

Comment:
Come think of it, this method should be private under a separate name, and taking a RegExp, a `type` and an `expected`.

Then we do something like this, which helps making the code more maintainable and makes errors a lot faster, since no string interpolation happens every time an error is thrown:
```ts
export function stringRegex(regex: RegExp) {
	return stringRegexValidator('s.string.regex', `expected ${regex}.test(expected) to be true`, regex);
}

export interface StringUuidOptions {
	version?: 1 | 3 | 4 | 5 | `${bigint}-${bigint}` | null;
	nullable?: boolean;
}

export function stringUuid({ version = 4, nullable = false }: StringUuidOptions = {}) {
	version ??= '1-5';
	const regex = new RegExp(`^(?:[0-9A-F]{8}-[0-9A-F]{4}-[${version}][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}${nullable ? '|00000000-0000-0000-0000-000000000000' : ''})$`, 'i');
	const expected = `expected to match UUID${typeof version === 'number' ? `v${version}` : ` in range of ${version}`}`;
	return stringRegexValidator('s.string.uuid', expected, regex);
}
```

Obviously, the methods in the validator should use the options interface, and just pass it throught.

Reply:
Oh yeah, and `NIL_UUID` should be removed, but we can probably do `version = 0` for exclusively null UUID, since it has a null version, wdyt @favna @vladfrangu